patches/server/1062-Add-ShulkerDuplicateEvent.patch

```diff
@@ -0,0 +1,26 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Chase Henderson <henderson.chase@gmail.com>
+Date: Fri, 5 Jan 2024 03:50:10 -0500
+Subject: [PATCH] Add ShulkerDuplicateEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/monster/Shulker.java b/src/main/java/net/minecraft/world/entity/monster/Shulker.java
+index b73dac8f68041f8a2e0752d70cc9d08b5cfd1cde..f2931e0e100bcae09601cae6c79816e562453927 100644
+--- a/src/main/java/net/minecraft/world/entity/monster/Shulker.java
++++ b/src/main/java/net/minecraft/world/entity/monster/Shulker.java
+@@ -485,6 +485,15 @@ public class Shulker extends AbstractGolem implements VariantHolder<Optional<Dye
+                     entityshulker.setVariant(this.getVariant());
+                     entityshulker.moveTo(vec3d);
+                     this.level().addFreshEntity(entityshulker, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.BREEDING); // CraftBukkit - the mysteries of life
++                    // Paper start - Shulker spawn event
++                    if (!entityshulker.isRemoved()) {
++                        io.papermc.paper.event.entity.ShulkerDuplicateEvent event = new io.papermc.paper.event.entity.ShulkerDuplicateEvent(entityshulker.getBukkitEntity(), this.getBukkitEntity());
++                        event.callEvent();
++                        if (event.isCancelled()) {
```

Comment:
this can all be just new Event().call I think, also, generally events are fired before something is added, why do you call it after addFreshEntity?